File: packages/category/general/domain/package-name/csharp/files/src/Service.cs
using System;
using System.Threading.Tasks;

namespace {{package_name | pascal_case}};

/// <summary>
/// {{package_description}}
/// </summary>
public class Service
{
    private readonly ILogger<Service> _logger;

    public Service(ILogger<Service> logger)
    {
        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
    }

    /// <summary>
    /// Example method that demonstrates basic functionality
    /// </summary>
    /// <param name="input">Sample input parameter</param>
    /// <returns>Sample result</returns>
    public async Task<string> ProcessAsync(string input)
    {
        _logger.LogInformation("Processing input: {Input}", input);

        try
        {
            // Your business logic here
            var result = await Task.FromResult($"Processed: {input}");
            
            _logger.LogInformation("Successfully processed input");
            return result;
        }
        catch (Exception ex)
        {
            _logger.LogError(ex, "Error processing input: {Input}", input);
            throw;
        }
    }

    /// <summary>
    /// Example synchronous method
    /// </summary>
    /// <param name="value">Sample value</param>
    /// <returns>Modified value</returns>
    public int MultiplyByTwo(int value)
    {
        _logger.LogDebug("Multiplying {Value} by 2", value);
        return value * 2;
    }
}

File: packages/category/general/domain/package-name/go/pkg/{{package_name}}/service.go
package {{package_name}}

import (
	"context"
	"fmt"

	"github.com/sirupsen/logrus"
)

// Service provides {{package_description}}
type Service struct {
	logger *logrus.Logger
}

// NewService creates a new Service instance
func NewService(logger *logrus.Logger) *Service {
	return &Service{
		logger: logger,
	}
}

// Process processes the input and returns a result
func (s *Service) Process(ctx context.Context, input string) (string, error) {
	s.logger.WithFields(logrus.Fields{
		"input": input,
	}).Info("Processing input")

	select {
	case <-ctx.Done():
		return "", ctx.Err()
	default:
		// Your business logic here
		result := fmt.Sprintf("Processed: %s", input)
		
		s.logger.Info("Successfully processed input")
		return result, nil
	}
}

// MultiplyByTwo multiplies the input by 2
func (s *Service) MultiplyByTwo(value int) int {
	s.logger.WithField("value", value).Debug("Multiplying by 2")
	return value * 2
}
